### PR TITLE
Add order events for discounting order/order_lines.

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds the order events connected to the discount action applied to the order/single order line.

**FYI. All fixme tags will be handled in separate PR**

https://app.clickup.com/t/2549495/SALEOR-1877

# Impact

* [x ] New migrations
* [ x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1875 New model Discount connected with Order](https://app.clickup.com/t/2549495/SALEOR-1875)

File: saleor/graphql/order/types.py [hunk 1]
```diff
@@ -48,10 +51,65 @@
...
+        old_value=discount_obj.get("old_value"),
+        old_amount=old_amount,
+    )
+
+
+class OrderDiscount(graphene.ObjectType):
+    value_type = graphene.Field(
+        DiscountValueTypeEnum,
+        required=True,
+        description="Type of the discount: fixed or percent",
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:79]: 👮‍♀️ 
```suggestion
        description="Type of the discount: fixed or percent.",
```

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -48,10 +51,65 @@
...
+    )
+    value = PositiveDecimal(
+        required=True,
+        description="Value of the discount. Can store fixed value or percent value",
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:83]: 👮‍♀️ 
```suggestion
        description="Value of the discount. Can store fixed value or percent value.",
```

File: saleor/graphql/order/types.py [hunk 3]
```diff
@@ -48,10 +51,65 @@
...
+
+class OrderEventDiscountObject(OrderDiscount):
+    old_value_type = graphene.Field(
+        DiscountValueTypeEnum,
+        required=False,
+        description="Type of the discount: fixed or percent",
+    )
+    old_value = PositiveDecimal(
+        required=False,
+        description="Value of the discount. Can store fixed value or percent value",
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:99]:
```suggestion
        description="Value of the discount. Can store fixed value or percent value.",
```

File: saleor/graphql/order/types.py [hunk 4]
```diff
@@ -48,10 +51,65 @@
...
+    )
+    reason = graphene.String(
+        required=False, description="Explanation for the applied discount."
+    )
+    amount = graphene.Field(Money, description="Returns amount of discount.")
+
+
+class OrderEventDiscountObject(OrderDiscount):
+    old_value_type = graphene.Field(
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:92]: Do we need these `old` prefixes?

File: saleor/graphql/order/types.py [hunk 5]
```diff
@@ -48,10 +51,65 @@
...
+        DiscountValueTypeEnum,
+        required=False,
+        description="Type of the discount: fixed or percent",
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:95]:
```suggestion
        description="Type of the discount: fixed or percent.",
```

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/order/types.py (hunk 4 above)
Comment (korycins) [reply saleor/graphql/order/types.py:93]: We need to store the normal values and the old values. The dashboard uses them to display the current value and the old value of the discount.

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**